packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-questions.ts

```diff
@@ -92,7 +92,7 @@ export async function askBucketNameQuestion(context: $TSContext, defaultValues:
   const message = 'Provide bucket name:';
   const onErrorMsg =
     'Bucket name can only use the following characters: a-z 0-9 - and should have minimum 3 character and max of 47 character';
-  const validator = and([alphanumeric(), minLength(3), maxLength(47)], onErrorMsg);
+  const validator = and([matchRegex('Input can only contain the following characters: a-z 0-9', /^[a-z0-9-]+$/),minLength(3), maxLength(47)], onErrorMsg);
```

Comment:
```suggestion
  const validator = and([matchRegex('Input can only contain lowercase characters, numbers, and hyphens', /^[a-z0-9-]+$/),minLength(3), maxLength(47)], onErrorMsg);
```

Reply:
Although this is more readable , it sort of increases the scope of the validator function . It would be correct to say: 
"Input can only contain lowercase characters, numbers, and hyphens. It cannot start or end with hyphens and should be between 3 and 47 chars long".

Reply:
another option is "Input can only contain the characters: a to z numbers: 0 to 9 and optionally - as a separator"

Reply:
Actually it should use the onErrorMsg variable instead of a custom string.

Reply:
Actually the validate function ignores the message if an onError is passed into a compound statement. 
In const validator = and([matchRegex('Input can only contain lowercase characters, numbers, and hyphens', /^[a-z0-9-]+$/),minLength(3), maxLength(47)], onErrorMsg);
The first error-string is ignored if an error is encountered. Hence the change may not be of consequence.